orchagent/policerorch.cpp

```diff
@@ -123,119 +142,395 @@ void PolicerOrch::doTask(Consumer &consumer)
+                m_syncdPolicers[key].policerOid);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to remove policer %s, rv:%d",
+                    key.c_str(), status);
+            return task_process_status::task_need_retry;
+        }
+
+        SWSS_LOG_NOTICE("Removed policer %s", key.c_str());
+        m_syncdPolicers.erase(key);
+        m_policerRefCounts.erase(key);
+    }
+
+    return task_process_status::task_success;
+}
+
+bool PolicerOrch::isStormControlPolicer(string policer_name)
+{
+    /*
+     * Temporary logic to identify storm-control policer
+     * To update the PolicerTable structure to include
+     * policer_type
+     */
+    if (policer_name[0] == '_')
+        return true;
+
+    return false;
+}
+
+/*Handler for "PORT_STORM_CONTROL" table*/
+task_process_status PolicerOrch::handlePortStormControlTable(Consumer& consumer)
+{
+    SWSS_LOG_ENTER();
+
+    KeyOpFieldsValuesTuple tuple = consumer.m_toSync.begin()->second;
+
+    string key = kfvKey(tuple);
+    string op = kfvOp(tuple);
+
+    /*<interface_name1>:<storm_type1>, <interface_name1>:<storm_type2>, ... (OR)
+      <interface_name1>:<storm_type1>, <interface_name2>:<storm_type1>, ...*/
+    vector<string> storm_control_keys = tokenize(key, list_item_delimiter);
+    for (string storm_key : storm_control_keys)
+    {
+        std::size_t delimiter = storm_key.find_first_of("|");
+        if (delimiter == std::string::npos)
+        {
+            SWSS_LOG_ERROR("Unable to parse key %s",storm_key.c_str());
+            /*continue here as there can be more interfaces*/
+            continue;
+        }
+        const auto interface_name = storm_key.substr(0,delimiter);
+        const auto storm_type = storm_key.substr(delimiter+1);
+        Port port;
+
+        /*Only proceed for Ethernet interfaces*/
+        if (strncmp(interface_name.c_str(), ETHERNET_PREFIX, strlen(ETHERNET_PREFIX)))
+        {
+            SWSS_LOG_ERROR("%s: Unsupported / Invalid interface %s",
+                    storm_type.c_str(), interface_name.c_str());
+            /*continue here as there can be more interfaces*/
+            continue;
+        }
+
+        if (!gPortsOrch->getPort(interface_name, port))
+        {
+            SWSS_LOG_ERROR("Failed to apply storm-control %s to port %s. Port not found",
+                    storm_type.c_str(), interface_name.c_str());
+            /*continue here as there can be more interfaces*/
+            continue; 
+        }
 
-        auto key = kfvKey(tuple);
-        auto op = kfvOp(tuple);
+        /*Policer Name: _<interface_name>_<storm_type>*/
+        const auto storm_policer_name = std::string("_").append(interface_name).append("_").append(storm_type);
 
         if (op == SET_COMMAND)
         {
             // Mark the opeartion as an 'update', if the policer exists.
             bool update = m_syncdPolicers.find(key) != m_syncdPolicers.end();
```

Comment:
the key to the map is the policer name which can be storm_policer_name (_interface_stormtype) or regular policer name. 
When list of interfaces is input, it would be split into vector of <interface|storm_type> and parsed individually.

Reply:
Fixed to use the 'storm_policer_name' as key